src/main/java/org/prebid/server/bidder/conversant/ConversantBidder.java

```diff
@@ -130,97 +129,105 @@ private static App updateApp(App app, String siteId) {
-                .bidfloor(extBidfloor != null ? extBidfloor : imp.getBidfloor())
-                .tagid(extTagId != null ? extTagId : imp.getTagid())
-                .secure(shouldChangeSecure ? extSecure : imp.getSecure())
-                .video(impVideo != null && impBanner == null ? modifyVideo(impVideo, impExt) : impVideo)
+                .bidfloor(impExt.getBidfloor())
+                .tagid(impExt.getTagId())
+                .secure(getSecure(imp, impExt))
+                .banner(modifyBanner(banner, impExt.getPosition()))
+                .video(video != null && banner == null ? modifyVideo(video, impExt) : video)
                 .build();
     }
 
+    private static Integer getSecure(Imp imp, ExtImpConversant impExt) {
+        final Integer extSecure = impExt.getSecure();
+        final Integer impSecure = imp.getSecure();
+
+        return extSecure != null && (impSecure == null || impSecure == 0) ? extSecure : impSecure;
+    }
+
     private static Banner modifyBanner(Banner impBanner, Integer extPosition) {
-        return impBanner == null || extPosition == null
-                ? impBanner
+        return impBanner == null
+                ? null
                 : impBanner.toBuilder()
-                .pos(AD_POSITIONS.contains(extPosition) ? extPosition : null)
+                .pos(isValidPosition(extPosition) ? extPosition : null)
                 .build();
     }
 
     private static Video modifyVideo(Video video, ExtImpConversant impExt) {
         final List<String> extMimes = impExt.getMimes();
-        final Integer extMaxduration = impExt.getMaxduration();
+        final Integer extMaxDuration = impExt.getMaxduration();
         final Integer extPosition = impExt.getPosition();
-        final List<Integer> extProtocols = impExt.getProtocols();
-        final List<Integer> extApi = impExt.getApi();
         return video.toBuilder()
-                .mimes(extMimes != null ? extMimes : video.getMimes())
-                .maxduration(extMaxduration != null ? extMaxduration : video.getMaxduration())
-                .pos(makePosition(extPosition, video.getPos()))
-                .api(makeApi(extApi, video.getApi()))
-                .protocols(makeProtocols(extProtocols, video.getProtocols()))
+                .mimes(CollectionUtils.isNotEmpty(extMimes) ? extMimes : video.getMimes())
+                .maxduration(extMaxDuration != null ? extMaxDuration : video.getMaxduration())
+                .pos(isValidPosition(extPosition) ? extPosition : null)
+                .api(makeApi(impExt.getApi(), video.getApi()))
+                .protocols(makeProtocols(impExt.getProtocols(), video.getProtocols()))
                 .build();
     }
 
-    private static Integer makePosition(Integer position, Integer videoPos) {
-        return isValidPosition(position) ? position : isValidPosition(videoPos) ? videoPos : null;
-    }
-
     private static boolean isValidPosition(Integer position) {
```

Comment:
Put this method above modify video

Reply:
Not sure in this case.
We have two subsequent methods `modifyBanner(..)` & `modifyVideo(..)` and the they use common functionality.